### PR TITLE
Upgrade sbt plugins. Fixes #2368

### LINKED ISSUE #2368: Failure to build 0.13.9 release with 0.13.8
Hi,
I downloaded 0.13.9 source tarball from github : https://github.com/sbt/sbt/archive/v0.13.9.tar.gz and tried to build it with sbt 0.13.8 from the web site : https://dl.bintray.com/sbt/native-packages/sbt/0.13.8/sbt-0.13.8.tgz .
But from those official tarballs, I'm unable to build sbt : 

```
[warn] There may be incompatibilities among your library dependencies.
[warn] Here are some of the libraries that were evicted:
[warn]  * com.typesafe.sbt:sbt-site:0.6.2 -> 0.7.1
[warn] Run 'evicted' to see detailed eviction warnings
java.lang.RuntimeException: Setting value cannot be null: {file:/build/sbt/sbt-0.13.9/}/*:gitCurrentBranch
    at scala.sys.package$.error(package.scala:27)
    at sbt.EvaluateSettings$INode.setValue(INode.scala:143)
    at sbt.EvaluateSettings$MixedNode.evaluate0(INode.scala:175)
    at sbt.EvaluateSettings$INode.evaluate(INode.scala:135)
    at sbt.EvaluateSettings$$anonfun$sbt$EvaluateSettings$$submitEvaluate$1.apply$mcV$sp(INode.scala:69)
    at sbt.EvaluateSettings.sbt$EvaluateSettings$$run0(INode.scala:78)
    at sbt.EvaluateSettings$$anon$3.run(INode.scala:74)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
    at java.lang.Thread.run(Thread.java:745)
[error] Setting value cannot be null: {file:/build/sbt/sbt-0.13.9/}/*:gitCurrentBranch
```

strace-ing the process it seems to be related to sbt-git looking for info in .git .
As the tarball is a release, .git is not present.
I stumble on this issue in sbt-git.
https://github.com/sbt/sbt-git/issues/36
Not sure if it was related to my problem. I wanted to force a dependency on sbt-git 0.7.1 (but I don't know the best way) so I changed : 
scala_2.10/sbt_0.13/com.typesafe.sbt/sbt-ghpages/ivy-0.5.1.xml and
scala_2.10/sbt_0.13/com.typesafe.sbt/sbt-ghpages/ivy-0.5.1.xml.original
especially : 

```
<dependency org="com.typesafe.sbt" name="sbt-git" rev="0.7.1" conf="compile->default(compile)" e:scalaVersion="2.10" e:sbtVersion="0.13"/>
```

I tried this on top of build.sbt but it didn't seem to help : 

```
libraryDependencies += "com.typesafe.sbt" % "sbt-git" % "0.7.1" force()
```

but modifying the xml file, I got the following log : 

```
[info] Resolving com.typesafe.sbt#sbt-git;0.7.1 ...
[warn]  module not found: com.typesafe.sbt#sbt-git;0.7.1
[warn] ==== local: tried
[warn]   /build/sbt/sbt-0.13.9/debian/sbt_home/.ivy2/local/com.typesafe.sbt/sbt-git/0.7.1/ivys/ivy.xml
[warn] ==== public: tried
[warn]   https://repo1.maven.org/maven2/com/typesafe/sbt/sbt-git/0.7.1/sbt-git-0.7.1.pom
[warn] ==== bintray-typesafe-ivy-releases: tried
[warn]   https://dl.bintray.com/typesafe/ivy-releases/com/typesafe/sbt/sbt-git/0.7.1/sbt-git-0.7.1.pom
[warn] ==== typesafe-ivy-releases: tried
[warn]   https://repo.typesafe.com/typesafe/ivy-releases/com.typesafe.sbt/sbt-git/0.7.1/ivys/ivy.xml
[warn] ==== typesafe-ivy-releases: tried
[warn]   https://repo.typesafe.com/typesafe/ivy-releases/com.typesafe.sbt/sbt-git/0.7.1/ivys/ivy.xml
[warn] ==== sonatype-snapshots: tried
[warn]   https://oss.sonatype.org/content/repositories/snapshots/com/typesafe/sbt/sbt-git/0.7.1/sbt-git-0.7.1.pom
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt.sxr/sxr_2.10/0.3.0/jars/sxr_2.10.jar ...
[info]  [SUCCESSFUL ] org.scala-sbt.sxr#sxr_2.10;0.3.0!sxr_2.10.jar (4277ms)
[info] downloading https://repo1.maven.org/maven2/org/scala-lang/scala-library/2.10.2/scala-library-2.10.2.jar ...
[info]  [SUCCESSFUL ] org.scala-lang#scala-library;2.10.2!scala-library.jar (6003ms)
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  ::          UNRESOLVED DEPENDENCIES         ::
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  :: com.typesafe.sbt#sbt-git;0.7.1: not found
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn] 
[warn]  Note: Unresolved dependencies path:
[warn]      com.typesafe.sbt:sbt-git:0.7.1 (/build/sbt/sbt-0.13.9/build.sbt#L10-11)
[warn]        +- org.scala-sbt:sbtroot:0.13.9
sbt.ResolveException: unresolved dependency: com.typesafe.sbt#sbt-git;0.7.1: not found
```

So, is it really a know problem of sbt-git ? what to do about it ? How come sbt-git 0.7.1 is not found ?
Thanks,
F.

### PR DESCRIPTION
**Pull request by dwijnand**
Tested manually:

```
  $ grep 'version.*:=' build.sbt
  version := "0.13.10-SNAPSHOT",
  $ git archive --prefix=sbt-0.13.10-SNAPSHOT/ -o ~/Desktop/sbt-0.13.10-SNAPSHOT.tgz HEAD
  $ cd ~/Desktop
  $ ls
  sbt-0.13.10-SNAPSHOT.tgz
  $ tar xfz sbt-0.13.10-SNAPSHOT.tgz
  $ cd sbt-0.13.10-SNAPSHOT/
  $ sbt info
  [info] Loading global plugins from /Users/dnw/.dotfiles/.sbt/0.13/plugins
  [info] Loading project definition from /Users/dnw/Desktop/sbt-0.13.10-SNAPSHOT/project
  [info] Updating {file:/Users/dnw/Desktop/sbt-0.13.10-SNAPSHOT/project/}sbt-0-13-10-snapshot-build...
  [info] Resolving org.fusesource.jansi#jansi;1.4 ...
  [info] Done updating.
  [warn] There may be incompatibilities among your library dependencies.
  [warn] Here are some of the libraries that were evicted:
  [warn]  * com.typesafe.sbt:sbt-site:0.8.1 -> 0.8.2
  [warn] Run 'evicted' to see detailed eviction warnings
  [info] Compiling 12 Scala sources to /Users/dnw/Desktop/sbt-0.13.10-SNAPSHOT/project/target/scala-2.10/sbt-0.13/classes...
  [info] This is sbt 0.13.9
  [info] The current project is {file:/Users/dnw/Desktop/sbt-0.13.10-SNAPSHOT/}sbtRoot 0.13.10-SNAPSHOT
  [info] The current project is built against Scala 2.10.6
  [info] Available Plugins: sbt.plugins.IvyPlugin, sbt.plugins.JvmPlugin, sbt.plugins.CorePlugin...
  [info] sbt, sbt plugins, and build definitions are using Scala 2.10.5
```

### CODE REVIEW

File: project/p.sbt [hunk 1]
```diff
@@ -1,18 +0,0 @@
-scalaVersion := "2.10.6"
-
-libraryDependencies ++= Seq(
-  "net.databinder" %% "dispatch-http" % "0.8.9",
-  "org.jsoup" % "jsoup" % "1.7.1"
```
Comment (dwijnand) [inline project/p.sbt:5]: these are the versions in the graph after updating the plugins below

**Comment by typesafe-tools** [comment]
Can one of the admins verify this patch?

File: project/bintray.sbt [hunk 2]
```diff
@@ -1,2 +0,0 @@
-addSbtPlugin("com.eed3si9n" % "bintray-sbt" % "0.3.0-a1934a5457f882053b08cbdab5fd4eb3c2d1285d")
-resolvers += Resolver.url("bintray-eed3si9n-sbt-plugins", url("https://dl.bintray.com/eed3si9n/sbt-plugins/"))(Resolver.ivyStylePatterns)
```
Comment (dwijnand) [inline project/bintray.sbt:2]: dropped, switching to released 0.3.0. I'm pretty sure the version above was the sha of @eed3si9n's changes while it was being pull request reviewed.

File: project/p.sbt [hunk 3]
```diff
@@ -1,18 +0,0 @@
...
-)
-
-addSbtPlugin("com.typesafe.sbt" % "sbt-site" % "0.7.1")
-
-resolvers += "jgit-repo" at "http://download.eclipse.org/jgit/maven"
```
Comment (dwijnand) [inline project/p.sbt:10]: no longer needed in sbt-git

File: project/plugins.sbt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+scalaVersion := "2.10.6"
+
+addSbtPlugin("com.eed3si9n" % "sbt-doge" % "0.1.5")
+addSbtPlugin("com.typesafe.sbt" % "sbt-ghpages" % "0.5.4")
+addSbtPlugin("com.typesafe.sbt" % "sbt-git" % "0.8.5")
```
Comment (dwijnand) [inline project/plugins.sbt:5]: Previously brought in transitively by sbt-ghpages. Added explicitly to use the latest release.

File: project/plugins.sbt [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+addSbtPlugin("com.typesafe.sbt" % "sbt-javaversioncheck" % "0.1.0")
+addSbtPlugin("com.typesafe.sbt" % "sbt-scalariform" % "1.3.0") // 1.6.0 is out but is a hard upgrade
```
Comment (dwijnand) [inline project/plugins.sbt:7]: I mean "a difficult upgrade"...

**Comment by eed3si9n** [comment]
LGTM

**Pull request merged**